### DISCUSSION TITLE
How to add a wrapper for the table

### DISCUSSION THREAD

**Discussion by RokasDie** [ueberdosis/tiptap#790] [General Questions]
Hello,

I need my tables to have css attribute ```overflow-x: auto```, after the content has been saved. I see that there is a div wrapper on the table while you are in editor mode: https://tiptap.scrumpy.io/tables
```


<div class="tableWrapper">
   <table style="min-width: 150px;">
      <colgroup>
         <col style="width: 100px;">
         <col>
         <col>
      </colgroup>
      <tbody>
         <tr>
            <th colspan="3" data-colwidth="100,0,0">
               <p>Wide header</p>
            </th>
         </tr>
         <tr>
            <td>
               <p>One</p>
            </td>
            <td>
               <p>Two</p>
            </td>
            <td>
               <p>Three</p>
            </td>
         </tr>
         <tr>
            <td>
               <p>Four</p>
            </td>
            <td>
               <p>Five</p>
            </td>
            <td>
               <p>Six</p>
            </td>
         </tr>
      </tbody>
   </table>
</div>

```

So after I submit my content this ```tableWrapper``` div disappears (together with ```colgroup``` element). How can I solve this? I need to have a wrapper div for my tables as this is the only way to add horizontal scrollbar just for them

**Comment by hanspagel** [comment]
I think you need to add your own Table node. See an example for the image node in #645. Hope that helps!

**Comment by LyDawei** [comment]
I ended up having to create my own TableView and use that as a stand in. I spent more time looking for solutions and reading documentation than I did implementing the solution. You can simply copy paste the Table View thats in the src and modify as needed to add your wrapper.

**Reply by gethari** [reply]
> 

Can you please share the source ?

**Reply by jakedolan** [reply]
This post has an example of how I did this. The lower code sample shows how to override the table renderHTML function.
https://github.com/ueberdosis/tiptap/issues/2041#issuecomment-950002510

**Reply by gethari** [reply]
Thanks I'll check it out
